src/plugins/es_ui_shared/server/license/license.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { i18n } from '@kbn/i18n';
+import {
+  Logger,
+  KibanaRequest,
+  KibanaResponseFactory,
+  RequestHandler,
+  RequestHandlerContext,
+} from 'src/core/server';
+
+/* eslint-disable @kbn/eslint/no-restricted-paths */
+import type {
+  LicenseType,
+  LicenseCheckState,
+} from '../../../../../x-pack/plugins/licensing/common/types';
+import type { LicensingPluginStart } from '../../../../../x-pack/plugins/licensing/server/types';
+/* eslint-enable @kbn/eslint/no-restricted-paths */
+
+interface SetupSettings {
+  pluginName: string;
+  logger: Logger;
+}
+
+interface StartSettings {
+  pluginId: string;
+  minimumLicenseType: LicenseType;
+  licensing: LicensingPluginStart;
+}
+
+export class License {
+  private pluginName?: string;
+  private logger?: Logger;
+  private licenseCheckState: LicenseCheckState = 'unavailable';
+  private licenseType?: LicenseType;
+
+  private _isEsSecurityEnabled: boolean = false;
+
+  setup({ pluginName, logger }: SetupSettings) {
+    this.pluginName = pluginName;
+    this.logger = logger;
+  }
+
+  start({ pluginId, minimumLicenseType, licensing }: StartSettings) {
+    licensing.license$.subscribe((license) => {
+      this.licenseType = license.type;
+      this.licenseCheckState = license.check(pluginId, minimumLicenseType!).state;
```

Comment:
To testers: you can hardcode a value here, e.g. 'invalid', to test the behavior under various conditions.